docs/design/keyboard-shortcuts.md

```diff
@@ -256,6 +256,69 @@ When using custom keyboard shortcuts on the web, some keyboard shortcuts that ar
+    })
+    .catch(function (ex) {
+        if (ex.code == "ActionNotFound") {
+            console.log("ActionId does not exist.");
+        } else if (ex.code == "InvalidShortcut") {
+            console.log("Shortcut combination is invalid.");
+        }
+    });
+```
+
+To populate a UI that specifies the add-in's shortcuts that the user can customize, call the [Office.actions.getShortcuts](/javascript/api/office/office.actions#getShortcuts) method. This method returns an object of type `[actionId:string]:string|null}`, where the `actionId`s are (1) all of the action IDs that are defined in the add-in's extended manifest JSON and (2) all the customized shortcuts registered for the user in the user's roaming settings. The values are the key combinations currently assigned to the actions. The following is an example.
+
+```javascript
+Office.actions.getShortcuts()
+    .then(function (userShortcuts) {
+       for (const action in userShortcuts) {
+           let shortcut = userShortcuts[action];
+           // In a real add-in you would typically present this information in the UI.
+           console.log(action + ": " + shortcut);
+       }
+    });
+
+```
+
+As described in [Avoid key combinations in use by other add-ins](#avoid-key-combinations-in-use-by-other-add-ins), the first time a user enters a key combination that is registered to multiple actions, the user is prompted to choose which action will be associated with the combination. This can be a dilemma for a user that wants more than one of the actions to have a shortcut. You can't prevent this from happening when the conflict is between your add-in and another add-in, but you can at least prevent conflicts between the *user's customization of one of your add-in's shortcuts* and a shortcut registered by an add-in or the host Office application. To prevent a conflict, you need to warn the user about it and give them the chance to change their custom key combination.
+
+To do this pass the user's custom key combinations, as an array of strings, to the [Office.actions.areShortcutsInUse](/javascript/api/office/office.actions#areShortcutsInUse) method. The method will return a report of which combinations are already in use in the form of an array of objects of type `{shortcut: string, inUse: boolean}`. The `shortcut` property is a key combination, such as "CTRL+SHIFT+1", and the `inUse` property is set to `true` if the combination is already registered to another action. For example, `[{shortcut: "CTRL+SHIFT+1", inUse: true}, {shortcut: "CTRL+SHIFT+2", inUse: false}]`. The following is an example in which some hypothetical user's custom key combinations are hardcoded. In an actual add-in, you programmatically get this information from the UI where the user has entered custom key combinations. Also, instead of just logging the results to the console, you would typically report the results to the UI and give the user a way to change key combinations that are already in use.
```

Comment:
We should also remove the reference to this "UI". Add-ins are not required to provide a UI but can do so if they want to.